storage/remote/queue_manager.go

```diff
@@ -214,6 +196,25 @@ func newQueueManagerMetrics(r prometheus.Registerer) *queueManagerMetrics {
 	return m
 }
 
+func (m *queueManagerMetrics) unregister() {
```

Comment:
would be nice if we could add slices of pointers (interface types? ) to the struct and just range loop through those for unregister?

Reply:
Hmmm... I wouldn't want consumers to need to use a map and casting, so would that be in addition to the metrics being defined on the struct?